src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -590,19 +590,16 @@ internal int BytesToWrite
         internal SerialStream(string portName, int baudRate, Parity parity, int dataBits, StopBits stopBits, int readTimeout, int writeTimeout, Handshake handshake,
             bool dtrEnable, bool rtsEnable, bool discardNull, byte parityReplace)
         {
-            int flags = NativeMethods.FILE_FLAG_OVERLAPPED;
-
-            if ((portName == null) || !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase))
+            if ((portName == null) ||
+                !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase) || 
+                !uint.TryParse(portName.Substring(3), out uint portNumber))
```

Comment:
This is an example of something that'll benefit from https://github.com/dotnet/corefx/issues/22403 to be able to avoid the string allocation just to parse the value out.